File: Swift-C++-Testsuite/test/arithmetic-operators/Substraction/Swift/UnarySub.swift
@main
struct MyMain {

    static func main() {

        let _ = unaryMinus(x: -1)
    }

    // The unary minus operator (-) inverts the current sign.
    // It performs the sign switch by calculating 0 - x
    static func unaryMinus(x: Int) -> Int {
        var a = x
        var b = -a
        return b
    }
}

File: Swift-C++-Testsuite/AutoTranspiledTests/try-1/UnarySub.cpp

#include <iostream>
 
struct MyMain {
 
    static void main(){
        std::cout << unaryMinus(-1) << std::endl;
    }
 
    // The unary minus operator (-) inverts the current sign.
    // It performs the sign switch by calculating 0 - x
    static int unaryMinus(int x) {
        int a = x;
        int b = 0 - a;
        return b;
    }
};
 
int main(){
    MyMain::main();
}